resources/js/Pages/Home.vue

```diff
@@ -138,64 +131,31 @@
           CdxButton,
           CdxIcon,
           CdxMessage,
-          InertiaHead,
-          TextArea,
+          TextareaHome,
+          InertiaHead
         },
         setup() {
+            const store = useStore();
+            const textareaInputValue = ref(store.lastSearchedIds);
+            
             return {
                 cdxIconDie,
-                cdxIconInfo
+                cdxIconInfo,
+                textareaInputValue
             };
         },
         methods: {
-            splitInput: function(): Array<string> {
-                return this.form.itemsInput.split( '\n' );
-            },
-            sanitizeArray: function(): Array<string> {
-                // this filter function removes all falsy values
-                // see: https://stackoverflow.com/a/281335/1619792
-                return this.splitInput().filter(x => x);
-            },
-            serializeInput: function(): string {
-                return this.sanitizeArray().join('|');
-            },
-            validate(): void {
-                this.validationError = null;
-
-                const rules = [{
-                    check: (ids: Array<string>) => ids.length < 1,
-                    type: 'warning',
-                    message: this.$i18n('item-form-error-message-empty')
-                },
-                {
-                    check: (ids: Array<string>) => ids.length > MAX_NUM_IDS,
-                    type: 'error',
-                    message: this.$i18n('item-form-error-message-max', MAX_NUM_IDS)
-                },
-                {
-                    check: (ids: Array<string>) => !ids.every(value => /^[Qq]\d+$/.test( value.trim() )),
-                    type: 'error',
-                    message: this.$i18n('item-form-error-message-invalid')
-                }];
-
-                const sanitized = this.sanitizeArray();
-
-                for(const {check, type, message} of rules){
-                    if(check(sanitized)){
-                        this.validationError = { type, message };
-                        return;
-                    }
-                }
-            },
             send(): void {
-                this.validate();
+                (this.$refs.textarea as InstanceType<typeof TextareaHome>).validate();
 
-                if(this.validationError) {
+                if((this.$refs.textarea as InstanceType<typeof TextareaHome>).validationError) {
                     return;
                 }
                 const store = useStore();
-                store.saveSearchedIds( this.form.itemsInput );
-                this.$inertia.get( '/results', { ids: this.serializeInput() } );
+                store.saveSearchedIds( this.textareaInputValue );
+                this.$inertia.get( '/results', 
+                    { ids: (this.$refs.textarea as InstanceType<typeof TextareaHome>).serializeInput() }
```

Comment:
I'd prefer it if we didn't directly call component methods. While this can be done with composition API using [`defineExpose`](https://vuejs.org/api/sfc-script-setup.html#defineexpose), the best practice here would be to emit an event with the already serialized input and use that event to save them to the component state. Then, this `send` method becomes much simpler, as it only needs to read the IDs from the state.